php-frontend/src/main/java/org/sonar/php/tree/impl/declaration/ClassDeclarationTreeImpl.java

```diff
@@ -77,10 +78,23 @@ protected ClassDeclarationTreeImpl(
     this.closeCurlyBraceToken = closeCurlyBraceToken;
   }
 
+  /**
+   * @deprecated since SonarQube 9.7 - Use {@link #modifiersToken()} instead.
+   */
   @Nullable
   @Override
+  @Deprecated
   public SyntaxToken modifierToken() {
```

Comment:
My bad. You only have to deprecate the method on the API site. So ClassDeclarationTree interface is the place where to add the annotation and the comment.

Reply:
I was not sure if this should be done at the interface level only or also on implementation, I checked the code and found some example where it's done on both (FunctionCallTreeImpl, ...), but indeed sound legit to me to have it only on the interface. Thanks !